x-pack/plugins/security_solution/public/timelines/components/open_timeline/export_timeline/export_timeline.tsx

```diff
@@ -47,20 +44,24 @@ const ExportTimeline: React.FC<{
     }
   }, [onComplete]);
 
-  return (
-    <>
-      {exportedIds != null && isEnableDownloader && (
-        <GenericDownloader
-          data-test-subj="export-timeline-downloader"
-          exportSelectedData={getExportedData}
-          filename={`${i18n.EXPORT_FILENAME}.ndjson`}
-          ids={exportedIds}
-          onExportSuccess={onExportSuccess}
-          onExportFailure={onExportFailure}
-        />
-      )}
-    </>
-  );
+  useEffect(() => {
+    const downloadTimeline = async () => {
+      if (exportedIds?.length && isEnableDownloader) {
+        const result = await exportSelectedTimeline({ ids: exportedIds });
+        if (result instanceof Blob) {
+          downloadBlob(result, `${i18n.EXPORT_FILENAME}.ndjson`);
+          onExportSuccess(exportedIds.length);
+        } else {
+          onExportFailure();
+        }
+      }
+    };
+
+    downloadTimeline();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why is this linter disabled? We are trying to not disable this except for the rare occurrence that we need to do just `[]` for an effect. If we really need to disable this, I would put a comment about why it's being disabled. It's a shame we have it disabled in so many places in the code base, but we are trying to not disable it as much as possible.

Reply:
I added it to mimic the behavior of the old `<GenericDownloader />` component (https://github.com/elastic/kibana/pull/100554/files/afd419abf31a9f7ea3a63dc8fb55e64aa208eeec#diff-a3d882482407c7bd482ae952442eafe9c68a8a465215cea8b53148af28468ad6L102). 

But TBH, instead of the `<ExportTimeline />` component we should have an imperative `exportTimeline()` method. Currently, it works like that:

1. A user clicks `Export`
2. We call `setState` with `ids` under the hood
3. `ExportTimeline` reacts to the state change with a `useEffect`
4. Inside the `useEffect` we fetch  timeline and start download

IMO, steps 2 and 3 are completely redundant we can skip them and start data fetch and download imperatively.

Reply:
Thanks for the info, do what you have to do here. That includes if you want to delay the additional work and write up a Kibana ticket and add maybe a comment in the code that points to the Kibana ticket (for example) so future maintainers or the next one through will see it and be like, "Ah, this really should not be a React Component it looks like".

Reply:
Added comment and a ticket to refactor file downloading: https://github.com/elastic/kibana/issues/101571